app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/photo/PhotoTaskFragment.kt

```diff
@@ -59,7 +59,9 @@ class PhotoTaskFragment : AbstractTaskFragment<PhotoTaskViewModel>() {
   // Registers a callback to execute after a user captures a photo from the on-device camera.
   private var capturePhotoLauncher: ActivityResultLauncher<Uri> =
     registerForActivityResult(ActivityResultContracts.TakePicture()) { result: Boolean ->
-      externalScope.launch { if (result) viewModel.savePhotoTaskData(capturedPhotoUri) }
+      externalScope.launch {
+        if (result && isViewModelInitialized) viewModel.savePhotoTaskData(capturedPhotoUri)
+      }
```

Comment:
This will prevent the exception from getting captured but won't save the captured photo. Can we also relaunch the coroutine when the view model is initialized?

Reply:
Added a new approach.